### DISCUSSION TITLE
[Feature Request] Support remote signing for the sequencer

### DISCUSSION THREAD

**Discussion by kevin-satsuma** [ethereum-optimism/developers#528] [General]
This is a feature request for the sequencer to support remote signing.

Currently `op-batcher`, `op-proposer`, and `contracts-bedrock` all support remote signing via KMS.  My team currently uses KMS to manage/sign transactions for admin/batcher/proposer wallets.

The sequencer is the only component right now that does not support remote signing.  We have to follow a different process when configuring the sequencer wallet for new rollups.  My team is a RaaS provider(Alchemy) so having KMS manage the wallets really helps simplify chain setup.  Even though the sequencer wallet does not hold funds, it would still be valuable to support remote signing in the sequencer to allow for op-stack wallets to be managed the same way.

From a security standpoint, this would also remove the possibility of leaking the sequencer private key via viewing env vars and/or CLI arguments to `op-node`.

Thanks in advance!

**Comment by sbvegan** [comment]
Hey @kevin-satsuma, I've raised this internally and its already on the teams radar, but we do not have any timelines at the moment. Thank you for raising this feature request!

**Reply by kevin-satsuma** [reply]
Thank you so much!

**Comment by tynes** [comment]
There was a governance proposal to get a grant to build an open source service that can handle the remote signing. Cannot seem to be able to find the application right now.

**Reply by Chomtana** [reply]
Will this require modification to the Optimism mono repo codebase?
